lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -2192,17 +2192,40 @@ namespace Js
             return scriptContext->GetLibrary()->GetUndefined();
         }
 #else
-        int required = ValidateAndCanonicalizeTimeZone(tz->GetSz());
-        if (required > 0)
+        UErrorCode status = U_ZERO_ERROR;
+
+        // TODO(jahorto): Is this the list of timeZones that we want? How is this different from
+        // the other two enum values or ucal_openTimeZones?
+        UEnumeration *available = ucal_openTimeZoneIDEnumeration(UCAL_ZONE_TYPE_ANY, nullptr, nullptr, &status);
```

Comment:
>&status [](start = 103, length = 7)

Do we need to assert this status is valid before continuing to the following line that overwrites status?

Reply:
All ICU APIs ignore calls if the passed in status is already in a failure state, so if you don't care about intermediate failure states, you could actually not check for U_FAILURE(status) until the caller is otherwise about to return something. So, it was easier to me to assert the status once since the uenum_count operation won't have any negative side effects if the enum failed to open.

I can probably add that as a comment, though.

Reply:
If that's standard with other ICU methods, then I don't think it needs a comment here. I wasn't familiar with the in-out style status pointer, but it's a neat idea.

---
In reply to: [169468076](https://github.com/Microsoft/ChakraCore/pull/4703#discussion_r169468076) [](ancestors = 169468076)